frontend/src/lib/Quiver.test.ts

```diff
@@ -0,0 +1,899 @@
+ * limitations under the License.
+ */
+
+import { util } from "apache-arrow"
+import { Quiver } from "src/lib/Quiver"
+import {
+  // Types
+  CATEGORICAL,
+  DATETIME,
+  FLOAT64,
+  INT64,
+  RANGE,
+  UINT64,
+  UNICODE,
+  // Special cases
+  EMPTY,
+  MULTI,
+  STYLER,
+  DISPLAY_VALUES,
+  FEWER_COLUMNS,
+  DIFFERENT_COLUMN_TYPES,
+} from "src/lib/mocks/arrow"
+
+describe("Quiver", () => {
+  describe("Public methods", () => {
+    describe("Without Styler", () => {
+      const mockElement = { data: UNICODE }
+      const q = new Quiver(mockElement)
+
+      test("cssId", () => {
+        expect(q.cssId).toBeUndefined()
+      })
+
+      test("cssStyles", () => {
+        expect(q.cssStyles).toBeUndefined()
+      })
+
+      test("caption", () => {
+        expect(q.caption).toBeUndefined()
+      })
+
+      test("dimensions", () => {
+        expect(q.dimensions).toStrictEqual({
+          headerRows: 1,
+          headerColumns: 1,
+          dataRows: 2,
+          dataColumns: 2,
+          rows: 3,
+          columns: 3,
+        })
+      })
+    })
+
+    describe("With Styler", () => {
+      const mockElement = {
+        data: STYLER,
+        styler: {
+          uuid: "FAKE_UUID",
+          styles: "FAKE_CSS",
+          displayValues: DISPLAY_VALUES,
+          caption: "FAKE_CAPTION",
+        },
+      }
+      const q = new Quiver(mockElement)
+
+      test("cssId", () => {
+        expect(q.cssId).toEqual("T_FAKE_UUID")
+      })
+
+      test("cssStyles", () => {
+        expect(q.cssStyles).toEqual("FAKE_CSS")
+      })
+
+      test("caption", () => {
+        expect(q.caption).toEqual("FAKE_CAPTION")
+      })
+
+      test("dimensions", () => {
+        expect(q.dimensions).toStrictEqual({
+          headerRows: 1,
+          headerColumns: 1,
+          dataRows: 2,
+          dataColumns: 2,
+          rows: 3,
+          columns: 3,
+        })
+      })
+    })
+
+    describe("getCell", () => {
+      const mockElement = { data: UNICODE }
+      const q = new Quiver(mockElement)
+
+      test("blank cell", () => {
+        expect(q.getCell(0, 0)).toStrictEqual({
+          type: "blank",
+          cssClass: "blank",
+          content: "",
+        })
+      })
+
+      test("index cell", () => {
+        expect(q.getCell(1, 0)).toStrictEqual({
+          type: "index",
+          cssClass: "row_heading level0 row0",
+          cssId: undefined,
+          content: "i1",
+          contentType: "unicode",
+        })
+      })
+
+      test("columns cell", () => {
+        expect(q.getCell(0, 1)).toStrictEqual({
+          type: "columns",
+          cssClass: "col_heading level0 col0",
+          content: "c1",
+        })
+      })
+
+      test("data cell", () => {
+        expect(q.getCell(1, 1)).toStrictEqual({
+          type: "data",
+          cssClass: "data row0 col0",
+          cssId: undefined,
+          content: "foo",
+          contentType: "unicode",
+        })
```

Comment:
For this test, can we use non-equal row + column indices, to make sure that we're testing for accidental transposition in the function? (So like "getCell(1, 2)" or whatever)

Reply:
Done